### PR TITLE
Improve generic analytics API

### PR DESCRIPTION
**Pull request by millerdev**
- Add semantically named functions for different types of events.
- Add generic API for Google Analytics

I discussed this with Amelia and we decided on two types of analytics events: usage and workflow. See the documentation for the distinction. I tried to make a generic-ish API that does not mention the services behind it (currently Google Analytics and Kissmetrics), but I'm not particularly happy with it because the wrapper around GA is very thin: it exposes the arguments of the underlying engine. I'm open to suggestions if people think this is a bad idea or can think of a better way to implement it.

The motivation for this is that we have switched analytics backends in the past, and I'd prefer to have a stable API to call in Vellum so we don't need to go around to every call site when we switch backends.

@NoahCarnahan 
cc @sravfeyn @dimagi/team-commcare-hq

### CODE REVIEW

**Comment by NoahCarnahan** [comment]
One option would be to give the `usage` function the same signature as `workflow`, and then pass the additional google analytics event identifiers as properties. So a call would look something like this:
```javascript
analytics.usage("Form Builder", {
    "action": "Create Question",
    "label": "select",
   "value": "multi"
});
```
The disadvantage is is more typing to make the function call... but then again maybe the keys would be useful to remember what all the args are for.

**Comment by NoahCarnahan** [comment]
p.s. At some point we'll probably want to create two versions of `trackLink` as well.

**Comment by millerdev** [comment]
On giving the two functions the same signature, I thought about that but it has it's own disadvantages. For example, if someone misspells one of the property names it presumably won't get sent through to the underlying engine.

File: corehq/apps/style/templates/style/includes/analytics_all.html [hunk 1]
```diff
@@ -106,6 +129,9 @@
         }
     };
 
+    // DEPRECATED
+    window.analytics.track = window.analytics.workflow;
```
Comment (sravfeyn) [inline corehq/apps/style/templates/style/includes/analytics_all.html]: Is this alias?

**Comment by millerdev** [comment]
@NoahCarnahan are you waiting on something here? I'd prefer to not change the function signature(s) for now.

File: corehq/apps/style/templates/style/includes/analytics_all.html (hunk 1 above)
Comment (millerdev) [inline corehq/apps/style/templates/style/includes/analytics_all.html]: Yeah, it's for backward compatibility.

**Comment by NoahCarnahan** [comment]
Not waiting on anything, was just offline for the weekend.

**Pull request merged**